### ISSUE TITLE
Malformatted line in Blast0_0.txt & ERROR: Blast0_0.txt is corrupted

### ISSUE THREAD

**Issue by yuan102379** [davidemms/OrthoFinder#749]
Hi davidemms,

Thanks for your nice OrthoFinder!

Recently, I have met with a strange problem when running Orthofinder.

**The running log is provided as follows**:
OrthoFinder version 2.5.4 Copyright (C) 2014 David Emms

2022-10-06 19:13:29 : Starting OrthoFinder 2.5.4
80 thread(s) for highly parallel tasks (BLAST searches etc.)
80 thread(s) for OrthoFinder algorithm

Checking required programs are installed
----------------------------------------
Test can run "mcl -h" - ok
Test can run "mafft /home/TY/glutamine/2.genome_pep_fasta/OrthoFinder/Results_Oct06/WorkingDirectory//_dependencies_check/SimpleTest.fa" - ok
Test can run "FastTree /home/TY/glutamine/2.genome_pep_fasta/OrthoFinder/Results_Oct06/WorkingDirectory//_dependencies_check/SimpleTest.fa" - ok

Dividing up work for BLAST for parallel processing
--------------------------------------------------
2022-10-06 19:13:32 : Creating diamond database 1 of 18
2022-10-06 19:13:32 : Creating diamond database 2 of 18
2022-10-06 19:13:33 : Creating diamond database 3 of 18
2022-10-06 19:13:33 : Creating diamond database 4 of 18
2022-10-06 19:13:33 : Creating diamond database 5 of 18
2022-10-06 19:13:33 : Creating diamond database 6 of 18
2022-10-06 19:13:33 : Creating diamond database 7 of 18
2022-10-06 19:13:33 : Creating diamond database 8 of 18
2022-10-06 19:13:33 : Creating diamond database 9 of 18
2022-10-06 19:13:34 : Creating diamond database 10 of 18
2022-10-06 19:13:34 : Creating diamond database 11 of 18
2022-10-06 19:13:34 : Creating diamond database 12 of 18
2022-10-06 19:13:34 : Creating diamond database 13 of 18
2022-10-06 19:13:34 : Creating diamond database 14 of 18
2022-10-06 19:13:34 : Creating diamond database 15 of 18
2022-10-06 19:13:34 : Creating diamond database 16 of 18
2022-10-06 19:13:35 : Creating diamond database 17 of 18
2022-10-06 19:13:35 : Creating diamond database 18 of 18

Running diamond all-versus-all
------------------------------
Using 80 thread(s)
2022-10-06 19:13:35 : This may take some time....
2022-10-06 19:13:35 : Done 0 of 324
2022-10-06 19:27:09 : Done 100 of 324
2022-10-06 19:33:48 : Done 200 of 324
2022-10-06 19:41:21 : Done all-versus-all sequence search

Running OrthoFinder algorithm
-----------------------------
2022-10-06 19:41:22 : Initial processing of each species
Malformatted line in /home/TY/glutamine/2.genome_pep_fasta/OrthoFinder/Results_Oct06/WorkingDirectory/Blast0_0.txt
Offending line was:

Process Process-82:
Traceback (most recent call last):
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 529, in Worker_ProcessBlastHits
    WaterfallMethod.ProcessBlastHits(*args, d_pickle=d_pickle, qDoubleBlast=qDoubleBlast)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 516, in ProcessBlastHits
    Bij = blast_file_processor.GetBLAST6Scores(seqsInfo, blastDir_list, seqsInfo.speciesToUse[iSpecies], seqsInfo.speciesToUse[jSpecies], qDoubleBla>
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/blast_file_processor.py", line 65, in GetBLAST6Scores
    for row in blastreader:
_csv.Error: line contains NUL
ERROR: Blast0_0.txt is corrupted
ERROR: Error processing files Blast0_*
Malformatted line in /home/TY/glutamine/2.genome_pep_fasta/OrthoFinder/Results_Oct06/WorkingDirectory/Blast1_0.txt
Offending line was:

Process Process-83:
Traceback (most recent call last):
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 529, in Worker_ProcessBlastHits
    WaterfallMethod.ProcessBlastHits(*args, d_pickle=d_pickle, qDoubleBlast=qDoubleBlast)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 516, in ProcessBlastHits
    Bij = blast_file_processor.GetBLAST6Scores(seqsInfo, blastDir_list, seqsInfo.speciesToUse[iSpecies], seqsInfo.speciesToUse[jSpecies], qDoubleBla>
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/blast_file_processor.py", line 65, in GetBLAST6Scores
    for row in blastreader:
_csv.Error: line contains NUL
ERROR: Blast1_0.txt is corrupted
ERROR: Error processing files Blast1_*
Malformatted line in /home/TY/glutamine/2.genome_pep_fasta/OrthoFinder/Results_Oct06/WorkingDirectory/Blast2_0.txt
Offending line was:

Process Process-85:
Traceback (most recent call last):
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 529, in Worker_ProcessBlastHits
    WaterfallMethod.ProcessBlastHits(*args, d_pickle=d_pickle, qDoubleBlast=qDoubleBlast)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 516, in ProcessBlastHits
    Bij = blast_file_processor.GetBLAST6Scores(seqsInfo, blastDir_list, seqsInfo.speciesToUse[iSpecies], seqsInfo.speciesToUse[jSpecies], qDoubleBla>
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/blast_file_processor.py", line 65, in GetBLAST6Scores
    for row in blastreader:
_csv.Error: line contains NUL
ERROR: Blast2_0.txt is corrupted
ERROR: Error processing files Blast2_*
Malformatted line in /home/TY/glutamine/2.genome_pep_fasta/OrthoFinder/Results_Oct06/WorkingDirectory/Blast3_0.txt
Offending line was:

Process Process-84:
Traceback (most recent call last):
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 529, in Worker_ProcessBlastHits
    WaterfallMethod.ProcessBlastHits(*args, d_pickle=d_pickle, qDoubleBlast=qDoubleBlast)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 516, in ProcessBlastHits
    Bij = blast_file_processor.GetBLAST6Scores(seqsInfo, blastDir_list, seqsInfo.speciesToUse[iSpecies], seqsInfo.speciesToUse[jSpecies], qDoubleBla>
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/blast_file_processor.py", line 65, in GetBLAST6Scores
    for row in blastreader:
_csv.Error: line contains NUL
ERROR: Blast3_0.txt is corrupted
ERROR: Error processing files Blast3_*
Malformatted line in /home/TY/glutamine/2.genome_pep_fasta/OrthoFinder/Results_Oct06/WorkingDirectory/Blast4_0.txt
Offending line was:

Process Process-86:
Traceback (most recent call last):
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/TY/anaconda3/envs/orthofinder/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 529, in Worker_ProcessBlastHits
    WaterfallMethod.ProcessBlastHits(*args, d_pickle=d_pickle, qDoubleBlast=qDoubleBlast)
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 516, in ProcessBlastHits
    Bij = blast_file_processor.GetBLAST6Scores(seqsInfo, blastDir_list, seqsInfo.speciesToUse[iSpecies], seqsInfo.speciesToUse[jSpecies], qDoubleBla>
  File "/home/TY/anaconda3/envs/orthofinder/bin/scripts_of/blast_file_processor.py", line 65, in GetBLAST6Scores
    for row in blastreader:
_csv.Error: line contains NUL
ERROR: Blast4_0.txt is corrupted
ERROR: Error processing files Blast4_*
Malformatted line in /home/TY/glutamine/2.genome_pep_fasta/OrthoFinder/Results_Oct06/WorkingDirectory/Blast6_0.txt
Offending line was:

.....

Looking forward to your response & Thanks again for your help.

Yours,

Yuan

**Comment by Changle-cell** [comment]
Could you please tell me about how did you solve this problem? Thank you very much.

**Comment by hcli007** [comment]
I solved ERROR: Blast3_ 0.txt is corrupted

Here is my solution:

Create a virtual environment with conda
`conda create -n orthofinder`

install old version 
`conda install -c bioconda orthofinder=2.4.0`

run orthofinder
`nohup orthofinder -f Dataset -M msa -S diamond -T iqtree -t 16 -a 16 > orthofinder.log &`

**Comment by Jihong528** [comment]
I upgraded the diamond installed with OrthoFinder version 3.0.1b1, which is copyrighted (C) 2014 David Emms, and the error was resolved. I activated the conda environment named `of3_env`. Since the diamond version was 0.9 when OrthoFinder was installed, I manually updated it. Here's the command I used to download and install the updated diamond:

```bash
wget -c https://anaconda.org/bioconda/diamond/2.1.13/download/linux-64/diamond-2.1.13-h13889ed_0.tar.bz2
conda install diamond-2.1.13-h13889ed_0.tar.bz2
```

I hope this will prevent the error from reoccurring.
